recipes-kernel/linux/files/patches-5.10/0121-dts-fix-the-gpio-configuration.patch

```diff
@@ -0,0 +1,41 @@
+From 5a96b3a123e350b170c0eae82c84efdf7d0d3d1c Mon Sep 17 00:00:00 2001
+From: chao zeng <chao.zeng@siemens.com>
+Date: Tue, 3 Jan 2023 16:58:54 +0800
+Subject: [PATCH 121/121] dts: fix the gpio configuration
+
+For the nodes main_gpio0 and main_gpio1,
+when we add the property pinctrl-0, it would override the
+previous property not to add to the previous node.
+So it would lack some configurations.
```

Comment:
What exactly would be broken?

Changes should be folded into the original commit adding the m2 dts file.

Reply:
for the serial port(cp2102n), when switching the serial mode, it should be reset right now. so we use `switchserialmode` to switch the mode, it would not take effect right away.